Source/UnitTests/Common/BitFieldViewTest.cpp

```diff
@@ -41,6 +41,59 @@ enum TestNoScopeEnum
   Doritos,
 };
 
+template <typename host_t, typename field_t, std::size_t start, std::size_t width>
+struct BitField2
+{
+private:
+  host_t host;
+
+public:
+  auto operator()()
+  {
+    return Common::MakeBitFieldView<field_t, start, width, host_t>(host);  //
+  }
+  auto operator()() const
+  {
+    return Common::MakeBitFieldView<field_t, start, width, host_t>(host);  //
+  }
+};
+
+struct TestStructAlternate
+{
+  union
+  {
+    u64 hex;
+
+    BitField2<u64, u64, 0, 64> full_u64;  // spans whole storage
+    BitField2<u64, s64, 0, 64> full_s64;  // spans whole storage
+
+    BitField2<u64, u64, 9, 3> regular_field_unsigned;   // a plain bitfield
+    BitField2<u64, u64, 9, 3> regular_field_unsigned2;  // Just the very same bitfield again
+    BitField2<u64, s64, 9, 3> regular_field_signed;     // Same bitfield, but different sign
+
+    BitField2<u64, s64, 30, 4> at_dword_boundary;  // goes over the boundary of two u32 values
+
+    BitField2<u64, s64, 15, 1> signed_1bit;  // allowed values: -1 and 0
+
+    BitField2<u64, bool, 63, 1> flag;
+    BitField2<u64, TestBoolEnum, 63, 1> enum_flag;
+
+    BitField2<u64, TestEnum, 16, 2> enum_1;
+    BitField2<u64, TestEnum, 48, 2> enum_2;
+    BitField2<u64, TestNoScopeEnum, 32, 2> enum_3;
+  };
+
+  TestStructAlternate() = default;
+  constexpr TestStructAlternate(u64 val) : hex(val) {}
+};
+
+TEST(BitFieldView, TestStructAlternate)
+{
+  TestStructAlternate object{0};
+
+  object.regular_field_unsigned() = 9;
```

Comment:
Isn't that [UB](https://stackoverflow.com/questions/11373203/accessing-inactive-union-member-and-undefined-behavior)? object's constructor seems to set `hex` as its active member, is accessing/setting another member and its operator() legal?

Reply:
If this is UB, then the existing classic BitField is also UB.  I feel like accessing the storage via lvalue reference makes it not UB somehow, but I could be (probably am) wrong.

Reply:
This is getting into obscure corners of the C++ standard... You can access inactive union members as long as the accessed data is part of the "Common Initial Sequence" of that inactive member and the active member.

For the current BitField, this is *almost* fulfilled, since BitField always uses the same type for storage (ignoring the sign, which technically makes it UB).

That being said, active member rules are irrelevant for gcc/clang/msvc since each of those compilers explicitly use their freedom-of-implementation to make access into inactive members well-defined.